File: app/src/main/java/com/example/dzj/myreader/utils/BookUtil.java
package com.example.dzj.myreader.utils;

/**
 * Created by dzj on 2017/8/20.
 */

public class BookUtil {
    //封面宽度
    private int width;
    //高度
    private int height;
    //左右边距
    private int hPadding;
    //上下边距
    private int vPadding;

    public BookUtil(double width){
        this.width=(int)width;
        setValue();
    }
    private void setValue(){
        double mheight=width/0.8;
        height=(int)mheight;
        double mhPadding=width*0.16;
        hPadding=(int)mhPadding;
        double mvPadding=mheight*0.10;
        vPadding=(int)mvPadding;
    }
    public int getHeight() {
        return height;
    }

    public int gethPadding() {
        return hPadding;
    }

    public int getvPadding() {
        return vPadding;
    }

    public int getWidth() {
        return width;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public void sethPadding(int hPadding) {
        this.hPadding = hPadding;
    }

    public void setvPadding(int vPadding) {
        this.vPadding = vPadding;
    }

    public void setWidth(int width) {
        this.width = width;
    }
    public String toString(){
        return "width="+width+"\nheight="+height+"\nleftPadding="+hPadding+"\ntopPadding="+vPadding;
    }
}

File: app/src/main/java/com/example/dzj/myreader/utils/BookUtil.kt
package com.example.dzj.myreader.utils

/**
 * Created by dzj on 2017/8/20.
 */

class BookUtil(width: Double) {
    //封面宽度
    var width: Int = 0
    //高度
    var height: Int = 0
    //左右边距
    private var hPadding: Int = 0
    //上下边距
    private var vPadding: Int = 0

    init {
        this.width = width.toInt()
        setValue()
    }

    private fun setValue() {
        val mheight = width / 0.8
        height = mheight.toInt()
        val mhPadding = width * 0.16
        hPadding = mhPadding.toInt()
        val mvPadding = mheight * 0.10
        vPadding = mvPadding.toInt()
    }

    fun gethPadding(): Int {
        return hPadding
    }

    fun getvPadding(): Int {
        return vPadding
    }

    fun sethPadding(hPadding: Int) {
        this.hPadding = hPadding
    }

    fun setvPadding(vPadding: Int) {
        this.vPadding = vPadding
    }

    override fun toString(): String {
        return "width=$width\nheight=$height\nleftPadding=$hPadding\ntopPadding=$vPadding"
    }
}
